src/System.Data.SqlClient/src/System/Data/Sql/SqlNorm.cs

```diff
@@ -336,51 +326,68 @@ internal sealed class ShortNormalizer : Normalizer
     {
         internal override void Normalize(FieldInfo fi, object obj, Stream s)
         {
-            byte[] b = BitConverter.GetBytes((short)GetValue(fi, obj));
+            short value = (short)GetValue(fi, obj);
+            Span<byte> b = stackalloc byte[2];
+
             if (!_skipNormalize)
             {
-                Array.Reverse(b);
+                BinaryPrimitives.WriteInt16BigEndian(b, value);
                 b[0] ^= 0x80;
             }
-            s.Write(b, 0, b.Length);
+            else
+            {
+                BinaryPrimitives.WriteInt16LittleEndian(b, value);
+            }
+
+            s.Write(b);
```

Comment:
Does this stream override the span-based write?

Reply:
Below are all the callsites for udt normalization:


https://github.com/dotnet/corefx/blob/09f4f7ed59992bc7bff6fe42bc75ef5b08c8c4e2/src/System.Data.SqlClient/src/System/Data/SqlClient/SqlConnection.cs#L1653
This uses MemoryStream, which does overwrite the span based write


https://github.com/dotnet/corefx/blob/09f4f7ed59992bc7bff6fe42bc75ef5b08c8c4e2/src/System.Data.SqlClient/src/System/Data/Sql/SqlSer.cs#L28-L36
https://github.com/dotnet/corefx/blob/09f4f7ed59992bc7bff6fe42bc75ef5b08c8c4e2/src/System.Data.SqlClient/src/System/Data/Sql/SqlSer.cs#L189-L192
This one doesn't, but is an internal stream and trivial to implement. Will adjust in next commit.

https://github.com/dotnet/corefx/blob/d3942d4671919edb0cca6ddc1840190f524a809d/src/System.Data.SqlClient/src/Microsoft/SqlServer/Server/ValueUtilsSmi.cs#L2628


https://github.com/dotnet/corefx/blob/a10890f4ffe0fadf090c922578ba0e606ebdd16c/src/System.Data.SqlClient/src/Microsoft/SqlServer/Server/SmiSettersStream.cs

This one doesn't either, but same as above, trivial to implement.

Reply:
Turns out the changing the SmiSettersStream to override the Span based write is more complicated than initially though. I would have to add/change the interface https://github.com/dotnet/corefx/blob/e0ba7aa8026280ee3571179cc06431baf1dfaaac/src/System.Data.SqlClient/src/Microsoft/SqlServer/Server/ITypedSettersV3.cs

It is an internal interface, so no breaking changes to the outside, but not sure if I should do it.
If anyone gives me the go-ahead, I would refactor this to adjust to Span.